src/python/pants/backend/python/lint/black/rules.py

```diff
@@ -7,40 +7,48 @@
 from typing import Optional, Tuple
 
 from pants.backend.python.lint.black.subsystem import Black
-from pants.backend.python.lint.python_formatter import PythonFormatter
+from pants.backend.python.lint.python_formatter import PythonFmtConfigurations
 from pants.backend.python.rules import download_pex_bin, pex
 from pants.backend.python.rules.pex import (
     Pex,
     PexInterpreterConstraints,
     PexRequest,
     PexRequirements,
 )
+from pants.backend.python.rules.targets import PythonSources
 from pants.backend.python.subsystems import python_native_code, subprocess_environment
 from pants.backend.python.subsystems.subprocess_environment import SubprocessEncodingEnvironment
 from pants.engine.fs import Digest, DirectoriesToMerge, PathGlobs, Snapshot
 from pants.engine.isolated_process import FallibleProcessResult, Process, ProcessResult
+from pants.engine.objects import Collection
 from pants.engine.rules import UnionRule, named_rule, rule, subsystem_rule
 from pants.engine.selectors import Get
 from pants.option.global_options import GlobMatchErrorBehavior
 from pants.python.python_setup import PythonSetup
 from pants.rules.core import determine_source_files, strip_source_roots
 from pants.rules.core.determine_source_files import (
-    LegacyAllSourceFilesRequest,
-    LegacySpecifiedSourceFilesRequest,
+    AllSourceFilesRequest,
     SourceFiles,
+    SpecifiedSourceFilesRequest,
 )
-from pants.rules.core.fmt import FmtResult
-from pants.rules.core.lint import Linter, LintResult
+from pants.rules.core.fmt import FmtConfiguration, FmtConfigurations, FmtResult
+from pants.rules.core.lint import LinterConfigurations, LintResult
 
 
 @dataclass(frozen=True)
-class BlackFormatter(PythonFormatter):
-    pass
+class BlackConfiguration(FmtConfiguration):
+    required_fields = (PythonSources,)
+
```

Comment:
any reason for this newline?

Reply:
Convention I've been using in the Configuration code to separate the `required_fields` line from the declaration of each field in the config. It looks more natural when there are more fields:

https://github.com/pantsbuild/pants/blob/46c30898c093bd06035f4a311d6fb0a4d443a099/src/python/pants/backend/python/rules/pytest_runner.py#L47-L53